### PR TITLE
Manifest for plugin `2`

### PR DESCRIPTION
**Pull request by steadysupply**
Please see https://steady.supply/git/kubectl-2/

### CODE REVIEW

**Comment by steadysupply** [comment]
I signed it

**Comment by steadysupply** [comment]
I signed it

**Comment by ahmetb** [comment]
I'm afraid the name `2` won't be applicable due to our naming conventions.
Have you read the Naming Guide?

File: plugins/2.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    which specifes a namespace to work in, a label filter used to select pods
+    and, of course, the local and remote paths involved.
+
+    the local path is watched, and when a file changes, that file is streamed
+    to all the pods matching the label specification.
+  platforms:
+    - selector:
+        matchLabels:
+          os: darwin
+      uri: https://steady.supply/git/kubectl-2/release/v0.0.0/kubectl-2-v0.0.0-darwin-amd64.tar.gz
```
Comment (ahmetb) [inline plugins/2.yaml]: Would you consider using github + github releases?
In that case we have tooling to automate releases, as well as we track download stats of github release assets.

File: plugins/2.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: "2"
+spec:
+  version: "v0.0.0"
+  shortDescription: dumb file transfer to your labelled pods
+  homepage: https://steady.supply/git/kubectl-2/
```
[the remark this answers is not in the document]
Comment (steadysupply) [reply plugins/2.yaml]: `git clone https://steady.supply/git/kubectl-2/` should do it for you

File: plugins/2.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: "2"
```
Comment (ahmetb) [inline plugins/2.yaml]: We normally reject plugins with non-descriptive names.
https://github.com/kubernetes-sigs/krew/blob/master/docs/NAMING_GUIDE.md
If you can suggest an alternative, that would be preferable.

File: plugins/2.yaml (hunk 1 above)
Comment (steadysupply) [reply plugins/2.yaml]: I'd rather self-host. Release automation is a few lines of shell `:)`

File: plugins/2.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahmetb) [reply plugins/2.yaml]: How can a user find this out, since it's not Github?
This is not immediately obvious.

**Comment by ahmetb** [comment]
@steadysupply before you move on to addressing more issues, we should hash out the [naming](https://github.com/kubernetes-sigs/krew/blob/master/docs/NAMING_GUIDE.md), probably. For now we probably can't get this in with this name, but we're actively working on trying to allow developers to run their own `krew-index`es so they have full freedom. Until that happens, we'll probably ask you to come up with an alternative name.

File: plugins/2.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (steadysupply) [reply plugins/2.yaml]: You're right, it's not obvious. Thanks for pointing that out `:)`

I considered changing the URL by adding the commonly used `.git` suffix, but that would introduce redundancy (we would have eg. `git/foo.git`). Instead I've added a note at the very top of the page that explains the URL hosts a Git repository as well as an HTML page. If you refresh your browser you should see it.

Hopefully that should clear things up for any future visitors!

[the remark this answers is not in the document]
Comment (corneliusweig) [reply plugins/2.yaml]: This is very borderline to being opaque to users. Binary plugins that don't show their source code cannot be merged into krew-index. Although the homepage mentions how the source code can be retrieved, this is unnecessarily hard to find.

How about mirroring your own repo to a github account?

**Comment by steadysupply** [comment]
e2e52574313c27845cf8cdbbdda99ff87571f719 points to `v0.0.1` which adds features

 - `ignore` key in config, useful if your editor writes temporary files which don't need to be sent to pods (vim!)
 - new logging
   - instead of printing a line per transfer, the current active transfers are displayed as dots on a single line which are removed when the transfer completes (this means the user doesn't end up with reams of identical output)
   - switch to [`glog`](https://github.com/golang/glog) which will make it possible to make the program produce structured logs (so if the user wants reams of identical output, they can get it)

**Comment by steadysupply** [comment]
95fe1ac points to `v0.0.2` which adds features
 - specify shell available on containers
 - display is more fun

**Comment by ahmetb** [comment]
@steadysupply hello, we're not actually that interested in changelogs. :) the naming is probably more important than that for now.

**Comment by corneliusweig** [comment]
/close
in favor of #502

**Pull request closed without merging**